databuilder/databuilder/extractor/dashboard/apache_superset/__init__.py

```diff
@@ -0,0 +1,174 @@
+import abc
+from functools import reduce
+from typing import (
+    Any, Dict, Generator, Iterator, List, Tuple,
+)
+
+import requests
+from bs4 import BeautifulSoup
+from dateutil import parser
+from pyhocon import ConfigFactory, ConfigTree
+
+from databuilder.extractor.base_extractor import Extractor
+
+type_fields_mapping = List[Tuple[str, str, Any, Any]]
+
+
+class ApacheSupersetBaseExtractor(Extractor):
+    """
+    Base class to create extractors pulling any dashboard metadata from Apache Superset.
+    """
+    APACHE_SUPERSET_PROTOCOL = 'apache_superset_protocol'
+    APACHE_SUPERSET_HOST = 'apache_superset_host'
+    APACHE_SUPERSET_PORT = 'apache_superset_port'
+    APACHE_SUPERSET_USER = 'apache_superset_user'
+    APACHE_SUPERSET_PASSWORD = 'apache_superset_password'
+    APACHE_SUPERSET_PAGE_SIZE = 'apache_superset_page_size'
+    APACHE_SUPERSET_EXTRACT_PUBLISHED_ONLY = 'apache_superset_extract_published_only'
+
+    DASHBOARD_GROUP_NAME = 'dashboard_group_name'
+    DASHBOARD_GROUP_ID = 'dashboard_group_id'
+    DASHBOARD_GROUP_DESCRIPTION = 'dashboard_group_description'
+
+    PRODUCT = 'product'
+    CLUSTER = 'cluster'
+
+    DRIVER_TO_DATABASE_MAPPING = 'driver_to_database_mapping'
+
+    DEFAULT_DRIVER_TO_DATABASE_MAPPING = {
+        'postgresql': 'postgres',
+        'mysql+pymysql': 'mysql'
+    }
+
+    DATABASE_TO_CLUSTER_MAPPING = 'database_to_cluster_mapping'  # map superset dbs to preferred clusters
```

Comment:
the superset db id (e.g which id maps which DW)?

Reply:
yes, superset db id (from database model)